rhizome/postgres/lib/hugepages_setup.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+require_relative "../../common/lib/util"
+require "logger"
+
+class HugepagesSetup
+  def initialize(instance, logger)
+    @version, @cluster = instance.split("-", 2)
+    @logger = logger
+  end
+
+  def get_postgres_param(param)
+    Integer(r("sudo -u postgres /usr/lib/postgresql/#{@version}/bin/postgres -D /dat/#{@version}/data -c config_file=/etc/postgresql/#{@version}/#{@cluster}/postgresql.conf -C #{param.shellescape}"), 10)
+  end
+
+  def hugepage_info
+    meminfo = File.read("/proc/meminfo")
+    hugepages_count = Integer(meminfo[/^HugePages_Total:\s*(\d+)/, 1], 10)
+    hugepage_size_kib = Integer(meminfo[/^Hugepagesize:\s*(\d+)\s*kB/, 1], 10)
+    [hugepages_count, hugepage_size_kib]
+  end
+
+  def stop_postgres_cluster
+    r "sudo pg_ctlcluster stop #{@version} #{@cluster}", expect: [0, 2] # 2 is "not running"
```

Comment:
I'm a little puzzled that we use pg_ctlcluster commands rather than systemd directly, but, if that's what we do...